infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -0,0 +1,80 @@
+package org.corfudb.infrastructure.logreplication.replication.receive;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.logprotocol.SMREntry;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.TableDescriptors;
+import org.corfudb.runtime.CorfuStoreMetadata.TableMetadata;
+import org.corfudb.runtime.CorfuStoreMetadata.TableName;
+import org.corfudb.runtime.collections.CorfuRecord;
+import org.corfudb.runtime.collections.CorfuTable;
+import org.corfudb.util.serializer.ISerializer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.corfudb.util.serializer.ProtobufSerializer.PROTOBUF_SERIALIZER_CODE;
+
+/**
+ * A parent class for Sink side StreamsSnapshotWriter and LogEntryWriter, which contains some common
+ * utility methods that could be used in both snapshot sync and log entry sync.
+ */
+@Slf4j
+public abstract class SinkWriter {
+    // A CorfuRuntime passed down from LogReplicationSinkManager.
+    final CorfuRuntime rt;
+
+    // Registry table for verifying replicated entries from Source side.
+    final CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> registryTable;
+
+    // Limit the initialization of this class only to its children classes.
+    SinkWriter(CorfuRuntime rt) {
+        this.rt = rt;
+        // Instantiate TableRegistry and register serializers in advance to avoid opening
+        // registry table in log entry writer's transaction
+        this.registryTable = rt.getTableRegistry().getRegistryTable();
+    }
+
+    /**
+     * In Log Replication table entries are sent to Sink in form of OpaqueEntry. Registry table's entries need
+     * to be deserialized in advance to avoid overwriting records that already existed in Sink's registry table.
+     * In addition, as the registry table must be read subsequently, new entries need to add their serialization
+     * info before applying, which is lost if the OpaqueEntry is written directly.
+     *
+     * This method will filter out the entries already exist in Sink side registry table, and add serialization info
```

Comment:
you can add this line in the @return java docs

Reply:
Changed, thx!